homeassistant/components/fan/__init__.py

```diff
@@ -469,6 +476,9 @@ def speed(self) -> str | None:
     @property
     def percentage(self) -> int | None:
         """Return the current speed as a percentage."""
+        if self._attr_percentage is not None:
+            return self._attr_percentage
```

Comment:
This will not work, as `None` is a valid value for percentage. Instead, we should not set the attribute and check if it has been set explicitly.

```suggestion
        if hasattr(self, "_attr_percentage"):
            return self._attr_percentage
```